src/storage/src/types/sinks.rs

```diff
@@ -422,7 +420,7 @@ impl RustType<ProtoKafkaSinkConnection> for KafkaSinkConnection {
                 .map(|(k, v)| (k.clone(), v.into_proto()))
                 .collect(),
             topic: self.topic.clone(),
-            topic_prefix: self.topic_prefix.clone(),
+            topic_prefix: self.topic.clone(),
```

Comment:
asking out of curiosity: why don't we just remove/deprecate this from the proto definition?

Reply:
Same answer for both, basically: a halfhearted attempt to give a hard error if folks are still using the old feature, instead of silently doing the wrong thing.

But I don't think we really care, and we're certainly breaking a lot this week anyways! So I'll clean them both out.